sympy/solvers/solvers.py

```diff
@@ -360,7 +360,12 @@ def checksol(f, symbol, sol=None, **flags):
         if val.is_Rational:
             return val == 0
         if numerical and val.is_number:
-            return (abs(val.n(18).n(12, chop=True)) < 1e-9) is S.true
+            try:
+                test = abs(val.n(18).n(12, chop=True)) < 1e-9
+            except TypeError:
+                return False
```

Comment:
I don't think that we can just return `False` in this case.

Also if the `TypeError` is only supposed to be caught from the `<` operator then that should be the only code inside the `try` block.

The exception is being raised because:
```
    362         if numerical and val.is_number:
--> 363             return (abs(val.n(18).n(12, chop=True)) < 1e-9) is S.true
    364 
    365     if flags.get('warn', False):
    366         warnings.warn("\n\tWarning: could not verify solution %s." % sol)
    367     # returns None if it can't conclude
    368     # TODO: improve solution testing
    369 

ipdb> p numerical
True
ipdb> p val.is_number
True
ipdb> p val
nan
ipdb> p val < 1e-9
*** TypeError: Invalid NaN comparison
```
Maybe it would be better if `nan.is_number` returned False:
```python
In [8]: nan.is_number
Out[8]: True
```